python/cudf/cudf/core/column/categorical.py

```diff
@@ -228,6 +233,9 @@ def __init__(self, dtype, mask=None, size=None, offset=0, children=()):
             respectively
         """
         if size is None:
+            for child in children:
+                assert child.offset == 0
+                assert child.mask is None
```

Comment:
Should use `base_mask` here just to avoid unnecessary work:
```suggestion
                assert child.base_mask is None
```